docs/src/pages/components/data-grid/filtering/QuickFilteringGrid.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import { IconButton, TextField } from '@material-ui/core';
+import {
+  DataGrid,
+  GridToolbarColumnsButton,
+  GridToolbarContainer,
+  GridToolbarDensitySelector,
+  GridToolbarFilterButton,
+} from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import ClearIcon from '@material-ui/icons/Clear';
+import SearchIcon from '@material-ui/icons/Search';
+
+const QuickSearchToolbar = (props) => {
+  return (
+    <GridToolbarContainer
+      style={{
+        justifyContent: 'space-between',
+      }}
+    >
+      <div>
+        <GridToolbarColumnsButton />
+        <GridToolbarFilterButton />
+        <GridToolbarDensitySelector />
+      </div>
+      <TextField
+        value={props.value}
```

Comment:
We need to force the variant as it's standard in v4 and outlined in v5 by default.

```suggestion
        variant="standar"
        value={props.value}
```

Reply:
We can involve @danilo-leal to improve the look and feel for the new search input:

<img width="830" alt="Capture d’écran 2021-07-15 à 16 55 06" src="https://user-images.githubusercontent.com/3165635/125809462-b7b8e9c1-1265-4b65-804e-df90e9a32508.png">

Reply:
<img width="500" alt="Screen Shot 2021-07-16 at 00 22 39" src="https://user-images.githubusercontent.com/67129314/125886807-fbb50e7c-0570-4586-b833-5987c755efec.png">

Quick iteration on it. Just added a little bit of x-axis/top padding on the div wrapping the buttons and the search input + also made the input underline a little bit lighter. Those types of interventions already boost the look a bit :)

Could definitely go further if appropriated now.